### PR TITLE
[lodash] enhance _.capitalize to return a capitalized string type

### PR DESCRIPTION
**Pull request by Ha-limLee**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://lodash.com/docs#capitalize>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/lodash/common/string.d.ts [hunk 1]
```diff
@@ -29,19 +29,19 @@ declare module "../index" {
          * @param string The string to capitalize.
          * @return Returns the capitalized string.
          */
-        capitalize(string?: string): string;
+        capitalize<T extends string>(string?: T): Capitalize<T>;
```
Comment (aj-r) [inline types/lodash/common/string.d.ts:32]: Lodash capitalize not only capitalizes the first letter, it also concerts the rest to lowercase https://lodash.com/docs#capitalize. So I think the return type should be:

`Capitalize<Lowercase<T>>`

File: types/lodash/lodash-tests.ts [hunk 2]
```diff
@@ -6623,10 +6623,23 @@ fp.now(); // $ExpectType number
...
+    // @ts-expect-error cannot assign non string type
+    _.capitalize(123);
+
+    _("fred").capitalize(); // $ExpectType "Fred"
+    _("fred" as string).capitalize(); // $ExpectType Capitalize<string>
+    _(123).capitalize(); // $ExpectType never
+
+    _.chain("fred").capitalize(); // $ExpectType StringChain<"Fred">
+    _.chain("fred" as string).capitalize(); // $ExpectType StringChain<Capitalize<string>>
+    _.chain(123).capitalize(); // $ExpectType StringChain<never>
```
[the remark this answers is not in the document]
Comment (aj-r) [reply types/lodash/lodash-tests.ts:6637]: That's fair. Ideally I think we should disallow calling it with non-string chains entirely, but that may not be feasible so what you have is fine.

**Review by aj-r — CHANGES_REQUESTED** [review]

File: types/lodash/common/string.d.ts (hunk 1 above)
Comment (Ha-limLee) [reply types/lodash/common/string.d.ts]: Oh, I missed that. Thanks!

File: types/lodash/common/lang.d.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import _ = require("../index");
-// eslint-disable-next-line @definitelytyped/strict-export-declare-modifiers
+// eslint-disable-next-line @definitelytyped/strict-export-declare-modifiers, @typescript-eslint/no-unsafe-function-type
 type GlobalFunction = Function;
```
Comment (Ha-limLee) [inline types/lodash/common/lang.d.ts:3]: I got a lint error here, but I believe it's safe to ignore it.

image

**Review by aj-r — APPROVED** [review]

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by Ha-limLee** [comment]
Ready to merge

**Pull request merged**